db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2375,7 +2379,8 @@ void DBImpl::MaybeScheduleFlushOrCompaction() {
     return;
   }
 
-  while (bg_compaction_scheduled_ < bg_job_limits.max_compactions &&
+  while (bg_compaction_scheduled_ + bg_bottom_compaction_scheduled_ <
```

Comment:
At present, `MaybeScheduleFlushOrCompaction()` does not schedule bottommost compaction. Is this intended or maybe I am missing something?

Reply:
That's right. We schedule automatic compaction before we know what we're going to compact. That's because the LSM could look different by the time our scheduled compaction job actually executes. So we schedule automatic compactions to low-pri thread pool first. Only after picking a bottom level compaction we forward it to the bottom-pri thread pool: https://github.com/facebook/rocksdb/blob/d94932323a9cfcfcbf37d9cfd86e7306fdd02c32/db/db_impl/db_impl_compaction_flush.cc#L3222-L3244